pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -609,10 +619,22 @@ func handleInitVerifiedObjects(ctx *verifierContext) {
 func verifyImageStatusFromImageFile(imageFileName string,
 	size int64, pathname string) *types.VerifyImageStatus {
 
+	// filename might have two parts, separated by '.': digest and PathEscape(mediaType)
+	var (
+		mediaType string
+		digest    string
+	)
+	parts := strings.Split(imageFileName, ".")
+	digest = parts[0]
+	if len(parts) == 2 {
```

Comment:
By the way, can we set the type somehow heuristically if it lacks? To prevent the failure when updating to the new version with the fix.

This part will work only after we save the media type in the filename with the fixed version. When rebooting to the fixed version from the previous version without the fix - it would be nice to have somehow restoration logic or at least to let the user know that it's a kind of expected failure.

Reply:
> By the way, can we set the type somehow heuristically if it lacks? To prevent the failure when updating to the new version with the fix

We could try, but I suspect it is fraught with errors. We may try processing too large a file, or make a mistake, etc. It might just be cleaner to have this be a known error and fail for it, rather than trying to guess our way through.

Reply:
Can't we return an error and force the download/verification process again?

Reply:
I think that is valid. Or maybe return a validation error, and whatever happens afterwards is the normal flow. But where do we do that?

Reply:
As far as I understand, if we return the nil status here, no VerifierStatus will not be populated and published:
https://github.com/lf-edge/eve/blob/304dc155230b4244dc9506d36ecacfa7e8be5b19/pkg/pillar/cmd/verifier/verifier.go#L685

In this case the check here
https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/volumemgr/blob.go#L419
will not pass, and the BlobStatus will not be populated leading to, according to your comment here (https://github.com/lf-edge/eve/issues/3704#issue-2081568716), a regular fallback mechanism, that workers fine. So, I would test returning nil.

Reply:
I'll test this approach locally now.

Reply:
I've tested it with several iterations, and it looks like it works. In this case, the application does not fail to start after the reboot. So, let's return nil here.

Reply:
@OhmSpectator you mean returning nil when there is len(parts) != 2?
Or returning nil when the mediaType string is an empty string?
(I don't know whether some local https datastore will set the content-type header.)

Reply:
I mean returning nil if len(parts) != 2

Reply:
@OhmSpectator That means if we have an image in /persist/vault/verified/ from before an EVE update to these bits, we will redownload it. That should be fine for that corner case.
But do we end up garbage collecting that old image, or can it stick around forever?

Reply:
Good point! The purpose was to force the redownload, exactly. I forgot about the garbage file in this case.  I'll think about the proper cleanup. FYI, @deitch.

Reply:
@OhmSpectator go for it. I pushed out changes based on your other comments, so don't forget to `pull` before `commit && push`

Reply:
I've added a change to address the cleanup and the redownload process.